mycore-base/src/main/java/org/mycore/common/config/MCRConfigurableInstanceHelper.java

```diff
@@ -0,0 +1,202 @@
+package org.mycore.common.config;
+
+import org.mycore.common.MCRClassTools;
+import org.mycore.common.MCRException;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * Creates Objects which are preconfigured with properties.
+ *
+ * Example Configuration<br>
+
+ * @author Sebastian Hofmann
+ */
+class MCRConfigurableInstanceHelper {
+
+    /**
+     * Creates a configured instance of a class extending .
+     * After creating the method with the {@link MCRPostConstruction} annotation is called with the property as
+     * parameter.
+      *@param property the property which contains the class
+     * @return a not null instance of T
+     * @throws MCRConfigurationException if the property is not right configured.
+     */
+    public static <T> Optional<T> getInstance(String property) throws MCRConfigurationException {
```

Comment:
can we reduce visibility to protected
I guess developers should still usually call MCRConfiguration2.getInstanceOf()
Now we have 2 public methods in 2 different classes that do the same thing

Reply:
The Class is only package visible.